ml_mr/estimation/quantile_iv.py

```diff
@@ -290,6 +292,171 @@ def from_results(cls, dir_name: str) -> "QuantileIVEstimator":
         return cls(exposure_network, outcome_network, meta=meta, covars=covars)
 
 
+class QuantileIVLinearEstimator(MREstimatorWithUncertainty):
+    """
+    QuantileIV estimator that uses the linear inference strategy, which
+    consists of performing IV linear regression manually in the new
+    representation of the exposure learned by the outcome neural network.
+    """
+    def __init__(
+        self,
+        dataset: IVDataset,
+        parent: QuantileIVEstimator
+    ):
+        self.parent = parent
+        self.exposure_network = parent.exposure_network
+        self.outcome_network = parent.outcome_network
+        self.covars = parent.covars
+        self.pca = self.init_PCA()
+        self.dataset = dataset
+        self.h = self.linear_inference(dataset)
+
+    def iv_reg_function(
+        self,
+        x: torch.Tensor,
+        covars: Optional[torch.Tensor] = None,
+        alpha: float = 0.1
+    ):
+        ys, se_ys = self.h(x, covars)
+        lower_ci = ys + stats.norm.ppf(alpha/2) * se_ys
+        upper_ci = ys + stats.norm.ppf(1-alpha/2) * se_ys
+
+        return torch.hstack([lower_ci, ys, upper_ci]).view(
+            len(ys), 1, 3
```

Comment:
FYI you can use -1 to "fill in" dimensions:

```suggestion
            -1, 1, 3
```